### PR TITLE
add ram observations

### PR DESCRIPTION
**Pull request by christopherhesse**
Add RAM observations.

It looks like `reset()` used `self.get_screen()` instead of `self.em.get_screen()`, the crop is not supposed to apply to observations, right @endrift?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Supported platforms:
 
 - Windows 7, 8, 10
 - macOS 10.12 (Sierra), 10.13 (High Sierra), 10.14 (Mojave)
-- Linux
```
Comment (endrift) [inline README.md]: Should be a separate commit

File: examples/random_agent.py [hunk 2]
```diff
@@ -4,16 +4,18 @@
 import retro
 
 parser = argparse.ArgumentParser()
-parser.add_argument('game', help='the name or path for the game to run')
-parser.add_argument('state', nargs='?', help='the initial state file to load, minus the extension')
```
Comment (endrift) [inline examples/random_agent.py]: Why these changes? Definitely should be a separate commit.

File: retro/retro_env.py [hunk 3]
```diff
@@ -191,8 +206,10 @@ def step(self, a):
         if self.movie:
             self.movie.step()
         self.em.step()
-        self.img = ob = self.get_screen()
+        self.img = self.em.get_screen()
```
Comment (endrift) [inline retro/retro_env.py]: Crops are intentional.

File: retro/retro_env.py [hunk 4]
```diff
@@ -208,9 +225,11 @@ def reset(self):
             self.movie_id += 1
         if self.movie:
             self.movie.step()
-        self.img = ob = self.get_screen()
+        self.img = self.em.get_screen()
```
Comment (endrift) [inline retro/retro_env.py]: Ditto.

File: retro/retro_env.py (hunk 3 above)
Comment (christopherhesse) [reply retro/retro_env.py]: My bad, I thought it was inconsistent about crops, but I must have mis-read the code.

File: README.md (hunk 1 above)
Comment (christopherhesse) [reply README.md]: Removed

File: examples/random_agent.py (hunk 2 above)
Comment (christopherhesse) [reply examples/random_agent.py]: Removed

File: retro/retro_env.py (hunk 4 above)
Comment (christopherhesse) [reply retro/retro_env.py]: Fixed

**Comment by christopherhesse** [comment]
I fixed the comments but also refactored `_get_obs` into `_update_obs` since this means I only need to get the requested type of observation and also `_get_obs` seemed to be only called right after updating.

File: retro/retro_env.py [hunk 5]
```diff
@@ -191,8 +208,8 @@ def step(self, a):
         if self.movie:
             self.movie.step()
         self.em.step()
-        self.img = ob = self.get_screen()
```
Comment (endrift) [inline retro/retro_env.py]: `self.img` still needs to be updated for `env.render()` to work, further the check at the beginning of `env.step` will need to be updated.

Comment (christopherhesse) [reply retro/retro_env.py]: Good call, thanks for the review!

**Comment by JoeOsborn** [comment]
This is exciting!  I have some use cases that need both RAM and the framebuffer—would it be possible to get, say, the framebuffer appended onto the RAM snapshot, or a 2-tuple of the two types of data?

**Comment by endrift** [comment]
Yeah, it may make sense to add a `BOTH` type that gives a DictSpace.

**Comment by christopherhesse** [comment]
A Dict space makes the most sense, but I think we can save that for another issue.   The user should be able to call `env.get_ram()` whenever so that could be called manually or as part of a wrapper that exposes both the image and the ram.

**Comment by christopherhesse** [comment]
@endrift it seems I am not added to the travis-ci.org retro project, could you add me please?

**Comment by christopherhesse** [comment]
Seems to work now, thanks!
